src/main/java/tech_tasks/array/find/previous/IndexOfPreviousNumberViaLoop.java

```diff
@@ -0,0 +1,95 @@
+package tech_tasks.array.find.previous;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class IndexOfPreviousNumberViaLoop {
+
+    @Test
+    void emptyArrayTest() {
+        int[] array = {};
+        assertThrows(IllegalArgumentException.class, () -> findIndexOfPrevious(array));
+    }
+
+    @Test
+    void singleElementArrayTest() {
+        int[] array = {5};
+        assertThrows(IllegalArgumentException.class, () -> findIndexOfPrevious(array));
+    }
+
+    @Test
+    void twoElementsArrayTest() {
+        int[] array = {5, 10};
+        assertEquals(0, findIndexOfPrevious(array));
+    }
+
+    @Test
+    void allSameElementsTest() {
+        int[] array = {5, 5, 5};
+        assertEquals(1, findIndexOfPrevious(array));
```

Comment:
Why it's not failed with "Array should contain at least two different elements"?